### PR TITLE
Fix spelling issues, add flake8-spellcheck whitelist

### PR DESCRIPTION
**Pull request by asears**
Signed-off-by: asears <asears@users.noreply.github.com>

# I have made things!

Here's a PR to address spelling issues in the repo as I get familiar with the code.  I leveraged flake8-spellcheck and added a whitelist.txt list of terms in the repo.  Flake8-spellcheck could be added to the dependencies and CI build.

A few of the tests failed locally, likely due to running on my Windows machine.  

There's one word which I'm unclear as to what it should be changed to, "contant", 

Looking for someone to assist with any documentation or changelog changes for this, if required.

Proselint is another tool which could help with minor updates to the documentation & comments.

eg.

```plaintext
docs\pages\api\tutorial.rst:31:1: after_the_deadline.redundancy Redundancy. Use 'first' instead of 'First of all,'.
docs\pages\api\tutorial.rst:55:18: weasel_words.very Substitute 'damn' every time you're inclined to write 'very'; your editor will delete it and the writing will be just as it should be. Found once elsewhere.
docs\pages\api\tutorial.rst:103:1: after_the_deadline.redundancy Redundancy. Use 'first' instead of 'First of all,'.
docs\pages\api\tutorial.rst:127:1: misc.but No paragraph should start with a 'But'.
docs\pages\api\tutorial.rst:155:35: misc.but No paragraph should start with a 'But'.
```

## Checklist

- [X] I have double checked that there are no unrelated changes in this pull request (old patches, accidental config files, etc)
- [ ] I have created at least one test case for the changes I have made
- [ ] I have updated the documentation for the changes I have made
- [ ] I have added my changes to the `CHANGELOG.md`

## Related issues

Refs #1525

🙏 Please, if you or your company is finding wemake-python-styleguide valuable, help us sustain the project by sponsoring it transparently on https://opencollective.com/wemake-python-styleguide. As a thank you, your profile/company logo will be added to our main README which receives hundreds of unique visitors per day.

### CODE REVIEW

File: tests/test_regressions/test_regression112.py
Comment (sobolevn) [inline tests/test_regressions/test_regression112.py:7]: It should be `breaks` I guess 🤔

File: wemake_python_styleguide/constants.py
Comment (sobolevn) [inline wemake_python_styleguide/constants.py:100]: Should be `Confusables`

**Review by sobolevn — CHANGES_REQUESTED** [review]
Awesome work!

We do have a lot of problems with spelling 😆 

I have several questions:
1. Can you please setup `flake8-spellcheck` to work in the CI?
2. Is it possible to move `whitelist.txt` to `tests/` folder? So it won't polute the root dir

**Comment by sobolevn** [comment]
> contant

It can possibly mean:
- `constant` in one context
- or `content` in another context 🤔

File: tests/test_regressions/test_regression112.py
Comment (asears) [inline tests/test_regressions/test_regression112.py:41]:
```suggestion
    module = ast.parse(code_that_breaks)
```

File: tests/test_regressions/test_regression112.py
Comment (asears) [inline tests/test_regressions/test_regression112.py:7]:
```suggestion
code_that_breaks = '''
```

File: wemake_python_styleguide/constants.py
Comment (asears) [inline wemake_python_styleguide/constants.py:100]:
```suggestion
    # Confusables:
```

**Comment by asears** [comment]
> Awesome work!
> 
> We do have a lot of problems with spelling 😆
> 
> I have several questions:
> 
> 1. Can you please setup `flake8-spellcheck` to work in the CI?

Thanks!  I would wait until this PR is merged before putting into CI.  That way build will not fail due to any missed spell checks.  Might need another round of linting after merge.

Can we add separate Issue/PR for CI integration & doc updates?  I think there's some breaking changes with flake8-spellcheck that may prevent this, at least in short term, might also be some incompatibilities with wemake.  

Longer term an updated flake8-spellcheck and a modified proselint might be a good addition or extension to wemake package.  If flake8-spellcheck is installed it will automatically get picked up by flake8 command.

If this is added as a poetry dependency that could be all that is required to add to CI.  

> 2. Is it possible to move `whitelist.txt` to `tests/` folder? So it won't polute the root dir
Since it's not related to Python tests, should it still go in tests folder?

I've put in an issue with flake8-spellcheck based on another similar request in other repo to move the location of the file and change the name.  By default it will look for whitelist.txt in the root, though it's configurable.  Can add separate issue for that outside of this change?  

Otherwise need to specify ./flake8 --whitelist ./tests/whitelist.txt parameter.   I think that requires a change to setup.cfg, again outside of this PR to fix the spelling?

**Comment by asears** [comment]
> > contant
> 
> It can possibly mean:
> 
> * `constant` in one context
> * or `content` in another context 🤔

Can use ```suggestion ``` and pick one?  I couldn't figure that out either but it's just a test.

**Comment by sobolevn** [comment]
> Can we add separate Issue/PR for CI integration & doc updates? I think there's some breaking changes with flake8-spellcheck that may prevent this, at least in short term, might also be some incompatibilities with wemake.

Yes, sure! 

> Otherwise need to specify ./flake8 --whitelist ./tests/whitelist.txt parameter. 

Yes, this is fine! I will do it myself.

Thanks a lot for your time and efforts 👍 
Great work!

**Review by sobolevn — APPROVED** [review]

**Pull request merged**
